### PR TITLE
Drop NodeJS 8

### LINKED ISSUE #328: Drop support for NodeJS 8
### 📦 Proposal

Node.js 8 has reached EOL as stated on the [official](https://nodejs.org/en/about/releases/) website and I think it should be good to bump the minimum required Node version of this project to v10.

The full list of EOL versions can be seen [here](https://github.com/nodejs/Release).

The good thing is that we can support the `Console.Table` API for things like #233 and avoid extra overhead on making stuff backward compatible.

Cheers!

### PR DESCRIPTION
**Pull request by KingDarBoja**
## PR Checklist

-   [x] Addresses an existing issue: fixes #328 
-   [x] That issue was marked as [`status: accepting prs`](https://github.com/typescript-eslint/tslint-to-eslint-config/labels/status%3A%20accepting%20prs)

## Overview

This PR has few changes:
- [x] Set minimum engine to Node +10.
- [x] Update dependencies to latest available version.
- [x] Provide missing return types on most error messages after running `npm run eslint`.
- [x] Export message on no-shadow-variable converter

### CODE REVIEW

**Comment by KingDarBoja** [comment]
@JoshuaKGoldberg ping!

File: src/adapters/exec.stubs.ts [hunk 1]
```diff
@@ -3,8 +3,8 @@ export type CreateStubExecSettings = {
     stdout: string;
 };
 
-export const createStubExec = ({ stderr = "", stdout = "" } = {}) =>
+export const createStubExec = ({ stderr = "", stdout = "" } = {}): jest.Mock =>
```
Comment (JoshuaKGoldberg) [inline src/adapters/exec.stubs.ts]: 🙁 I have a preference for the way it was before: not redundantly declaring return types. Could you please undo these unrelated source changes?

File: src/rules/converters/tests/no-shadowed-variable.test.ts [hunk 2]
```diff
@@ -1,4 +1,8 @@
-import { convertNoShadowedVariable } from "../no-shadowed-variable";
+import {
+    convertNoShadowedVariable,
+    SELECTIVE_DISABLE_NOTICE,
+    UNDERSCORE_DISABLE_NOTICE,
+} from "../no-shadowed-variable";
```
Comment (JoshuaKGoldberg) [inline src/rules/converters/tests/no-shadowed-variable.test.ts]: Similarly, I don't see why these kinds of changes would be necessary to drop NodeJS 8 support. Let's revert.

File: src/adapters/exec.stubs.ts (hunk 1 above)
Comment (KingDarBoja) [reply src/adapters/exec.stubs.ts]: Apologies for the change, I do remember updating the dependencies that triggered these changes due to TSLint 6.0.0.

Should be good if you can provide a PR to fix those TSLint issues while I revert most of the changes done 🙈

**Comment by KingDarBoja** [comment]
@JoshuaKGoldberg I will close this PR and open a new one with only the CircleCI and the package.json engine changes.

But these stuff should be done on separate PRs:
- Bump dependencies
- TSLint rules (maybe migrate to ESLint?).

Once again, my apologies for wasting your time when I made this PR as I tend to put extra stuff on my PRs (I couldn't resist 🙈 ). But been learning to control my need for fix everything at once.

Cheers!

**Comment by JoshuaKGoldberg** [comment]
Haha that's a struggle I have to get through the hard way... at least several times a year 😹. Thanks!

> Bump dependencies

I sent a request to the typescript-eslint super-admins to install Renovate to take care of that. Good callout.

> TSLint rules

By which you mean.. updating to the newest typescript-eslint rules, since we're on ESLint? I'm up for discussing, but don't feel particularly passionate about them - feel free to open up issues for ones you'd like to enable.

**Comment by KingDarBoja** [comment]
> By which you mean.. updating to the newest typescript-eslint rules, since we're on ESLint? I'm up for discussing, but don't feel particularly passionate about them - feel free to open up issues for ones you'd like to enable.

Ah yeah, ESLint ones, I got confused too 🙈 Looks like I haven't been up to date with latest changes.
